src/main/java/com/google/sps/servlets/PostDataServlet.java

```diff
@@ -32,41 +40,65 @@
-        PreparedQuery results = datastore.prepare(query);
-        ArrayList<Post> posts = Post.queryToPosts(results, datastore);
+    // Queries Datastore with the college ID and receives posts such that the soonest events are shown first.
+    Query query = new Query(collegeId).addSort("timeSort", SortDirection.ASCENDING);
+    PreparedQuery results = datastore.prepare(query);
+    ArrayList<Post> posts = Post.queryToPosts(results, datastore);
 
-        // Prepares the relevant posts in JSON.
-        String json = listToJson(posts);
-        response.setContentType("application/json");
-        response.getWriter().println(json);
-    }
+    // Prepares the relevant posts in JSON.
+    String json = listToJson(posts);
+    response.setContentType("application/json");
+    response.getWriter().println(json);
+  }
+
+  /* On the POST command, serializes the request information into Post objects,
+   * stores the post as entity in Datastore.
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException { 
+    String collegeId = request.getParameter("collegeId");
+    Post newPost = new Post();
+    newPost.requestToPost(request, collegeId);
+    Entity newPostEntity = newPost.postToEntity();
+    datastore.put(newPostEntity);
+
+    notifyUsers(collegeId, newPost);
+
+    String redirectURL ="/find-events.html?" + "collegeid=" + collegeId;
+    response.sendRedirect(redirectURL);
+  }
 
-    /* On the POST command, serializes the request information into Post objects. */
-    /* Stores the post as entity in Datastore, with the collegeId as the kind. */
-    @Override
-    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException { 
-        String collegeId = request.getParameter("collegeId");
-        Post newPost = new Post();
-        newPost.requestToPost(request, collegeId);
-        Entity newPostEntity = newPost.postToEntity();
-        datastore.put(newPostEntity);
+  /**
+    * Send emails to all users associated with the specific college.
+    *
+    * @param collegeId unique id of a college
+    * @throws IOException
+    */
+  public void notifyUsers(String collegeId, Post newPost) throws IOException {
+    
+    // Find all users that attend the college.
+    Filter universityFilter = new FilterPredicate("university", FilterOperator.EQUAL, collegeId);
+    Query q = new Query("User").setFilter(universityFilter);
+    PreparedQuery pq = datastore.prepare(q);
 
-        String redirectURL ="/find-events.html?" + "collegeid=" + collegeId;
-        response.sendRedirect(redirectURL);
+    // Email the users to notify them that a new post has been added.
+    for (Entity user : pq.asIterable()) {
+      String email = user.getKey().getName().toString();
+      GmailConfiguration.sendEmail(email, Email.newPostSubject, Email.addNewPost(newPost));	
     }
+  }
```

Comment:
I'm wondering if the querying and sending of email should be handled by the email servlet?

Reply:
Great idea but emails don't have a servlet! However, I placed it in the configuration aspect!